tests/System.Text.Primitives.Tests/EncodingPerfTestsCoreCLR.cs

```diff
@@ -0,0 +1,180 @@
+                    output = Encoding.Convert(utf8, utf8, utf8Buffer);
+            }
+        }
+
+        [Benchmark]
+        [InlineData(DataLength, 0x0, Utf8OneByteLastCodePoint)]
+        [InlineData(DataLength, Utf8OneByteLastCodePoint + 1, Utf8TwoBytesLastCodePoint)]
+        [InlineData(DataLength, Utf8TwoBytesLastCodePoint + 1, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, Utf16HighSurrogateFirstCodePoint, Utf16LowSurrogateLastCodePoint)]
+        [InlineData(DataLength, 0x0, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.AlternatingASCIIAndNonASCII)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.MostlyASCIIAndSomeNonASCII)]
+        public void EncodeFromUtf16toUtf8CoreCLR(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf8 = Encoding.UTF8;
+            int utf8Length = utf8.GetByteCount(characters);
+            var utf8Buffer = new byte[utf8Length];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    utf8.GetBytes(characters, 0, characters.Length, utf8Buffer, 0);
+            }
+        }
+
+        [Benchmark]
+        [InlineData(DataLength, 0x0, Utf8OneByteLastCodePoint)]
+        [InlineData(DataLength, Utf8OneByteLastCodePoint + 1, Utf8TwoBytesLastCodePoint)]
+        [InlineData(DataLength, Utf8TwoBytesLastCodePoint + 1, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, Utf16HighSurrogateFirstCodePoint, Utf16LowSurrogateLastCodePoint)]
+        [InlineData(DataLength, 0x0, Utf8ThreeBytesLastCodePoint)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.AlternatingASCIIAndNonASCII)]
+        [InlineData(DataLength, 0, 0, SpecialTestCases.MostlyASCIIAndSomeNonASCII)]
+        public void EncodeFromUtf32toUtf8CoreCLR(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf32 = Encoding.UTF32;
+            Encoding utf8 = Encoding.UTF8;
+            int utf32Length = utf32.GetByteCount(characters);
+            var utf32Buffer = new byte[utf32Length];
+            utf32.GetBytes(characters, 0, characters.Length, utf32Buffer, 0);
+
+            int utf8Length = utf8.GetByteCount(characters);
+            var output = new byte[utf8Length];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    output = Encoding.Convert(utf32, utf8, utf32Buffer);
```

Comment:
So, are you suggesting go from UTF-32 to UTF-16 and then from UTF-16 to UTF-8 instead?
Should I do the same thing for the Corefxlab tests even though we have a (potentially) faster UTF-32 to UTF-8 direct encoder? I was trying to keep the comparison consistent. I think chaining the decoders will make the CoreCLR test slower.